pkg/volume/util/nested_volumes.go

```diff
@@ -51,8 +51,14 @@ func getNestedMountpoints(name, baseDir string, pod v1.Pod) ([]string, error) {
 				return fmt.Errorf("invalid container mount point %v", myMountPoint)
 			}
 			myMPSlash := myMountPoint + string(os.PathSeparator)
-			// The previously found nested mountpoint (or "" if none found yet)
-			prevNestedMP := ""
+			// The previously found nested mountpoints.
+			// NOTE: We can't simply rely on sort.Strings to have all the mountpoints sorted and
```

Comment:
They're still sorted (L45), but now we're checking at L71-77 if that directory was seen already.

For example, let's say we have ``\dir\nested\foo``, ``\dir\nested2``, ``\dir\nested``. This will be sorted to ``\dir\nested``, ``\dir\nested2``, ``\dir\nested\foo``. Being the first one, ``\dir\nested`` is added to ``retval``. The second one, doesn't match any previous mountpoint (``\dir\nested2`` doesn't have the  ``\dir\nested\`` prefix), it is then added to ``retval``. FInally, ``\dir\nested\foo`` has the prefix ``\dir\nested\``, so it's skipped.

The sort is still useful, it still makes sure that ``\dir\nested\foo`` is sorted after ``\dir\nested``, which will mean that ``\dir\nested\foo`` will be treated as an already seen mountpoint.